lib/chef/resource/gem_package.rb

```diff
@@ -55,6 +55,14 @@ class GemPackage < Chef::Resource::Package
       property :options, [ String, Hash, Array, nil ],
         description: "Options for the gem install, either a Hash or a String. When a hash is given, the options are passed to Gem::DependencyInstaller.new, and the gem will be installed via the gems API. When a String is given, the gem will be installed by shelling out to the gem command. Using a Hash of options with an explicit gem_binary will result in undefined behavior.",
         desired_state: false
+
+      def clear_sources?
+        if clear_sources.nil?
+          !!source
```

Comment:
this should really get moved to the provider since its not interface and is logic.

should document the magic here as well with a comment.

Reply:
If by document the magic you mean to leave a comment explaining what this is doing, I had intended to do that after I got confirmation that this is what we wanted 👍